x-vector/src/X/Data/Vector/Generic.hs

```diff
@@ -29,9 +44,71 @@ import           Data.Vector.Fusion.Stream.Monadic (Stream(..))
 import qualified Data.Vector.Fusion.Stream.Monadic as Stream
 
 import           Data.Vector.Generic as Generic
+import qualified Data.Vector.Generic.Mutable as MGeneric
+import qualified Data.Vector.Unboxed as Unboxed
 
 import           P hiding (mapMaybe)
 
+-- | The 'transpose' function transposes the rows and columns of its argument.
+--
+--   For example:
+--
+--   > transpose [[1,2,3],[4,5,6]] == [[1,4],[2,5],[3,6]]
```

Comment:
gah github! I didn't even see @amosr's overloaded lists comment until I did a hard refresh, seems like some comments come over the web socket and some don't, pretty flakey :/